pkg/ansible/generate.go

```diff
@@ -140,79 +144,243 @@ openshift_aws_create_s3: false
-[[ $key ]]: [[ $value ]]
+openshift_aws_clusterid: [[ .Name ]]
+openshift_aws_elb_basename: [[ .Name ]]
+openshift_aws_vpc_name: [[ .Name ]]
+[[ if .DefaultAMI ]]
+openshift_aws_ami: [[ .DefaultAMI ]]
 [[ end ]]
 `
+	masterVarsTemplate = `
+openshift_aws_ami_map:
+  master: [[ .AMIName ]]
 
-// TODO: these {{}} don't play nice in a go template, complete this when we get to processing
-// node groups dynamically
-/*
-# tags
-openshift_aws_master_group_tags:
-  Name: "{{ openshift_aws_clusterid }}-master"
-  host-type: master
-  sub-host-type: default
-
-openshift_aws_node_group_compute_tags:
-  Name: "{{ openshift_aws_clusterid }}-compute"
-  host-type: node
-  sub-host-type: compute
-
-openshift_aws_node_group_infra_tags:
-  Name: "{{ openshift_aws_clusterid }}-infra"
-  host-type: node
-  sub-host-type: infra
-*/
+openshift_aws_master_group_config:
+  # The 'master' key is always required here.
+  master:
+    instance_type: [[ .InstanceType ]]
+    volumes: "{{ openshift_aws_node_group_config_master_volumes }}"
+    health_check:
+      period: 60
+      type: EC2
+    min_size: 1
+    max_size: [[ .Size ]]
+    desired_size: [[ .Size ]]
+    wait_for_instances: True
+    termination_policy: "{{ openshift_aws_node_group_termination_policy }}"
+    replace_all_instances: "{{ openshift_aws_node_group_replace_all_instances }}"
+    iam_role: "{{ openshift_aws_iam_role_name }}"
+    policy_name: "{{ openshift_aws_iam_role_policy_name }}"
+    policy_json: "{{ openshift_aws_iam_role_policy_json }}"
+    elbs: "{{ openshift_aws_elb_name_dict['master'].keys()| map('extract', openshift_aws_elb_name_dict['master']) | list }}"
+`
+	infraVarsTemplate = `
+openshift_aws_ami_map:
+  infra: [[ .AMIName ]]
 
-const (
-	cloudProviderKindVar = "openshift_cloudprovider_kind"
-	awsClusterIDVar      = "openshift_aws_clusterid"
-	elbBasenameVar       = "openshift_aws_elb_basename"
-	awsVPCNameVar        = "openshift_aws_vpc_name"
-	awsRegionVar         = "openshift_aws_region"
-	awsAMIVar            = "openshift_aws_ami"
+openshift_aws_node_group_config:
+  infra:
+    instance_type: [[ .InstanceType ]]
+    volumes: "{{ openshift_aws_node_group_config_node_volumes }}"
+    health_check:
+      period: 60
+      type: EC2
+    min_size: 1
+    max_size: [[ .Size ]]
+    desired_size: [[ .Size ]]
+    termination_policy: "{{ openshift_aws_node_group_termination_policy }}"
+    replace_all_instances: "{{ openshift_aws_node_group_replace_all_instances }}"
+    iam_role: "{{ openshift_aws_iam_role_name }}"
+    policy_name: "{{ openshift_aws_iam_role_policy_name }}"
+    policy_json: "{{ openshift_aws_iam_role_policy_json }}"
+    elbs: "{{ openshift_aws_elb_name_dict['infra'].keys()| map('extract', openshift_aws_elb_name_dict['infra']) | list }}"
+
+openshift_aws_node_groups:
+- name: "{{ openshift_aws_clusterid }} infra group"
+  group: infra
+  tags:
+    host-type: node
+    sub-host-type: infra
+    runtime: docker
+
+openshift_aws_ami_map:
```

Comment:
This is duplicated from the beginning of `infraVarsTemplate`.

Reply:
thx, will remove